### PR TITLE
[1.16.*] Client classes mappings

### PR DESCRIPTION
**Pull request by Matyrobbrt**
Mapps the parameters in classes from the following packages:
net.minecraft.client.renderer.item
net.minecraft.client.renderer.texture
net.minecraft.client.resources
net.minecraft.client.resources.language
net.minecraft.client.resources.metadata.*
net.minecraft.client.server
net.minecraft.client.searchtree
net.minecraft.client.sounds
net.minecraft.client.tutorial
net.minecraft.client.gui.screens

net.minecraft.commands

net.minecraft.advancements
net.minecraft.advancements.criterion

and some other random classes

### CODE REVIEW

File: data/net/minecraft/client/gui/screens/LanguageSelectScreen.mapping [hunk 1]
```diff
@@ -3,6 +3,10 @@ CLASS net/minecraft/client/gui/screens/LanguageSelectScreen
 		COMMENT Reference to the LanguageManager object.
 	FIELD packSelectionList Lnet/minecraft/client/gui/screens/LanguageSelectScreen$LanguageSelectionList;
 		COMMENT The List GuiSlot object reference.
+	METHOD <init> (Lnet/minecraft/client/gui/screens/Screen;Lnet/minecraft/client/Options;Lnet/minecraft/client/resources/language/LanguageManager;)V
+		ARG 1 scrren
```
Comment (sciwhiz12) [inline data/net/minecraft/client/gui/screens/LanguageSelectScreen.mapping]: Typo:
```suggestion
		ARG 1 screen
```

File: data/net/minecraft/client/gui/screens/LanguageSelectScreen.mapping [hunk 2]
```diff
@@ -3,6 +3,10 @@ CLASS net/minecraft/client/gui/screens/LanguageSelectScreen
...
+		ARG 2 options
+		ARG 3 langManager
```
Comment (sciwhiz12) [inline data/net/minecraft/client/gui/screens/LanguageSelectScreen.mapping]: Parameter names should follow the name of the field they're assigned to.
```suggestion
		ARG 3 languageManager
```

File: data/net/minecraft/client/gui/screens/LanguageSelectScreen.mapping [hunk 3]
```diff
@@ -12,6 +16,8 @@ CLASS net/minecraft/client/gui/screens/LanguageSelectScreen
 		METHOD renderBackground (Lcom/mojang/blaze3d/vertex/PoseStack;)V
 			ARG 1 matrixStack
 		CLASS Entry
+			METHOD <init> (Lnet/minecraft/client/gui/screens/LanguageSelectScreen$LanguageSelectionList;Lnet/minecraft/client/resources/language/LanguageInfo;)V
+				ARG 2 lang
```
Comment (sciwhiz12) [inline data/net/minecraft/client/gui/screens/LanguageSelectScreen.mapping]: Parameter names should follow the name of the field they're assigned to.
```suggestion
				ARG 2 language
```

File: data/net/minecraft/client/gui/screens/OptionsScreen.mapping [hunk 4]
```diff
@@ -1,10 +1,17 @@
 CLASS net/minecraft/client/gui/screens/OptionsScreen
 	FIELD options Lnet/minecraft/client/Options;
 		COMMENT Reference to the GameSettings object.
```
Comment (sciwhiz12) [inline data/net/minecraft/client/gui/screens/OptionsScreen.mapping]: As a side-note, could you please remove this javadoc?

File: data/net/minecraft/client/gui/screens/PresetFlatWorldScreen.mapping [hunk 5]
```diff
@@ -1,6 +1,10 @@
 CLASS net/minecraft/client/gui/screens/PresetFlatWorldScreen
 	FIELD parent Lnet/minecraft/client/gui/screens/CreateFlatWorldScreen;
 		COMMENT The parent GUI
+	METHOD <init> (Lnet/minecraft/client/gui/screens/CreateFlatWorldScreen;)V
+		ARG 1 parent
+	METHOD fromString (Lnet/minecraft/core/Registry;Ljava/lang/String;Lnet/minecraft/world/level/levelgen/flat/FlatLevelGeneratorSettings;)Lnet/minecraft/world/level/levelgen/flat/FlatLevelGeneratorSettings;
+		ARG 0 biome
```
Comment (sciwhiz12) [inline data/net/minecraft/client/gui/screens/PresetFlatWorldScreen.mapping]: This should be named `biomeRegistry`, as `biome` implies it is a `Biome`.
```suggestion
		ARG 0 biomeRegistry
```

File: data/net/minecraft/client/gui/screens/PresetFlatWorldScreen.mapping [hunk 6]
```diff
@@ -14,6 +18,13 @@ CLASS net/minecraft/client/gui/screens/PresetFlatWorldScreen
 		ARG 1 minecraft
 		ARG 2 width
 		ARG 3 height
+	METHOD save (Lnet/minecraft/core/Registry;Lnet/minecraft/world/level/levelgen/flat/FlatLevelGeneratorSettings;)Ljava/lang/String;
+		ARG 0 biome
```
Comment (sciwhiz12) [inline data/net/minecraft/client/gui/screens/PresetFlatWorldScreen.mapping]: This should be named `biomeRegistry`, as `biome` implies it is a `Biome`.
```suggestion
		ARG 0 biomeRegistry
```

File: data/net/minecraft/client/gui/screens/WinScreen.mapping [hunk 7]
```diff
@@ -1,6 +1,13 @@
...
+		ARG 2 onFinished
 	METHOD render (Lcom/mojang/blaze3d/vertex/PoseStack;IIF)V
 		ARG 1 matrixStack
 		ARG 2 mouseX
 		ARG 3 mouseY
 		ARG 4 partialTicks
+	METHOD renderBg (IIF)V
+		ARG 1 x
+		ARG 2 y
+		ARG 3 partialTicks
```
Comment (sciwhiz12) [inline data/net/minecraft/client/gui/screens/WinScreen.mapping]: `partialTick` would be more approriate, as this is a `float` ranging from 0.0 to 1.0 that encodes a fraction (or part) of a tick since the last tick.
```suggestion
		ARG 3 partialTick
```

File: data/net/minecraft/client/gui/screens/achievement/StatsScreen.mapping [hunk 8]
```diff
@@ -1,6 +1,9 @@
 CLASS net/minecraft/client/gui/screens/achievement/StatsScreen
 	FIELD isLoading Z
 		COMMENT When true, the game will be paused when the gui is shown
+	METHOD <init> (Lnet/minecraft/client/gui/screens/Screen;Lnet/minecraft/stats/StatsCounter;)V
+		ARG 1 lastScreen
+		ARG 2 statsCounter
```
Comment (sciwhiz12) [inline data/net/minecraft/client/gui/screens/achievement/StatsScreen.mapping]: Parameter names should follow the name of the field they're assigned to.
```suggestion
		ARG 2 stats
```

File: data/net/minecraft/client/player/AbstractClientPlayer.mapping [hunk 9]
```diff
@@ -1,4 +1,7 @@
 CLASS net/minecraft/client/player/AbstractClientPlayer
+	METHOD <init> (Lnet/minecraft/client/multiplayer/ClientLevel;Lcom/mojang/authlib/GameProfile;)V
+		ARG 1 level
```
Comment (sciwhiz12) [inline data/net/minecraft/client/player/AbstractClientPlayer.mapping]: Parameter names should follow the name of the field they're assigned to.
```suggestion
		ARG 1 clientLevel
```

File: data/net/minecraft/client/player/LocalPlayer.mapping [hunk 10]
```diff
@@ -13,6 +13,14 @@ CLASS net/minecraft/client/player/LocalPlayer
 		COMMENT The last yaw value which was transmitted to the server, used to determine when the yaw changes and needs to be re-transmitted
 	FIELD zLast D
 		COMMENT The last Z position which was transmitted to the server, used to determine when the Z position changes and needs to be re-transmitted
+	METHOD <init> (Lnet/minecraft/client/Minecraft;Lnet/minecraft/client/multiplayer/ClientLevel;Lnet/minecraft/client/multiplayer/ClientPacketListener;Lnet/minecraft/stats/StatsCounter;Lnet/minecraft/client/ClientRecipeBook;ZZ)V
+		ARG 1 minecraft
+		ARG 2 level
+		ARG 3 connection
+		ARG 4 stats
+		ARG 5 recipeBppk
```
Comment (sciwhiz12) [inline data/net/minecraft/client/player/LocalPlayer.mapping]: Typo:
```suggestion
		ARG 5 recipeBook
```

File: data/net/minecraft/client/renderer/CubeMap.mapping [hunk 11]
```diff
@@ -1,4 +1,6 @@
 CLASS net/minecraft/client/renderer/CubeMap
+	METHOD <init> (Lnet/minecraft/resources/ResourceLocation;)V
+		ARG 1 location
```
Comment (sciwhiz12) [inline data/net/minecraft/client/renderer/CubeMap.mapping]: This should be a more specific name rather than a generic `location`. Perhaps `textureLocationBase`?

File: data/net/minecraft/client/renderer/PostPass.mapping [hunk 12]
```diff
@@ -1,4 +1,9 @@
 CLASS net/minecraft/client/renderer/PostPass
+	METHOD <init> (Lnet/minecraft/server/packs/resources/ResourceManager;Ljava/lang/String;Lcom/mojang/blaze3d/pipeline/RenderTarget;Lcom/mojang/blaze3d/pipeline/RenderTarget;)V
+		ARG 1 resourceMnager
```
Comment (sciwhiz12) [inline data/net/minecraft/client/renderer/PostPass.mapping]: Typo:
```suggestion
		ARG 1 resourceManager
```

File: data/net/minecraft/client/renderer/Rect2i.mapping [hunk 13]
```diff
@@ -1,4 +1,9 @@
 CLASS net/minecraft/client/renderer/Rect2i
+	METHOD <init> (IIII)V
+		ARG 1 xPOs
```
Comment (sciwhiz12) [inline data/net/minecraft/client/renderer/Rect2i.mapping]: Typo:
```suggestion
		ARG 1 xPos
```

File: data/net/minecraft/client/renderer/texture/Stitcher.mapping [hunk 14]
```diff
@@ -1,4 +1,8 @@
 CLASS net/minecraft/client/renderer/texture/Stitcher
+	METHOD <init> (III)V
+		ARG 1 mipLevel
+		ARG 2 maxWidth
+		ARG 3 maxHeight
```
Comment (sciwhiz12) [inline data/net/minecraft/client/renderer/texture/Stitcher.mapping]: These seem to be in the wrong order.
```suggestion
		ARG 1 maxWidth
		ARG 2 maxHeight
		ARG 3 mipLevel
```

File: data/net/minecraft/client/resources/DirectAssetIndex.mapping [hunk 15]
```diff
@@ -1,3 +1,5 @@
 CLASS net/minecraft/client/resources/DirectAssetIndex
+	METHOD <init> (Ljava/io/File;)V
+		ARG 1 assetsDir
```
Comment (sciwhiz12) [inline data/net/minecraft/client/resources/DirectAssetIndex.mapping]: Parameter names should follow the name of the field they're assigned to.
```suggestion
		ARG 1 assetsDirectory
```

File: data/net/minecraft/client/resources/SkinManager.mapping [hunk 16]
```diff
@@ -1,4 +1,8 @@
 CLASS net/minecraft/client/resources/SkinManager
+	METHOD <init> (Lnet/minecraft/client/renderer/texture/TextureManager;Ljava/io/File;Lcom/mojang/authlib/minecraft/MinecraftSessionService;)V
+		ARG 1 manager
+		ARG 2 skinsDir
```
Comment (sciwhiz12) [inline data/net/minecraft/client/resources/SkinManager.mapping]: Parameter names should follow the name of the field they're assigned to.
```suggestion
		ARG 2 skinsDirectory
```

File: data/net/minecraft/client/resources/language/LanguageInfo.mapping [hunk 17]
```diff
@@ -0,0 +1,6 @@
+CLASS net/minecraft/client/resources/language/LanguageInfo
+	METHOD <init> (Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Z)V
+		ARG 1 code
+		ARG 2 region
+		ARG 3 name
+		ARG 4 biDirectional
```
Comment (sciwhiz12) [inline data/net/minecraft/client/resources/language/LanguageInfo.mapping]: Parameter names should follow the name of the field they're assigned to.
```suggestion
		ARG 4 bidirectional
```

File: data/net/minecraft/client/resources/metadata/animation/AnimationFrame.mapping [hunk 18]
```diff
@@ -0,0 +1,6 @@
+CLASS net/minecraft/client/resources/metadata/animation/AnimationFrame
+	METHOD <init> (I)V
+		ARG 1 index
+	METHOD <init> (II)V
+		ARG 1 indxe
```
Comment (sciwhiz12) [inline data/net/minecraft/client/resources/metadata/animation/AnimationFrame.mapping]: Typo:
```suggestion
		ARG 1 index
```

File: data/net/minecraft/client/searchtree/ReloadableIdSearchTree.mapping [hunk 19]
```diff
@@ -12,3 +14,6 @@ CLASS net/minecraft/client/searchtree/ReloadableIdSearchTree
 		COMMENT If the query does not contain a <code>:</code>, then only {@link #byName} is searched" if it does contain a colon, both {@link #byName} and {@link #byId} are searched and the results are merged using a {@link MergingIterator}.
 		COMMENT @return A list of all matching items in this search tree.
 		ARG 1 searchText
+	CLASS IntersectionIterator
+		METHOD <init> (Ljava/util/Iterator;Ljava/util/Iterator;Ljava/util/Comparator;)V
+			ARG 3 order
```
Comment (sciwhiz12) [inline data/net/minecraft/client/searchtree/ReloadableIdSearchTree.mapping]: Parameter names should follow the name of the field they're assigned to.
```suggestion
			ARG 3 orderT
```

File: data/net/minecraft/client/searchtree/ReloadableSearchTree.mapping [hunk 20]
```diff
@@ -10,3 +15,6 @@ CLASS net/minecraft/client/searchtree/ReloadableSearchTree
 		COMMENT If the query does not contain a <code>:</code>, then only {@link #byName} is searched" if it does contain a colon, both {@link #byName} and {@link #byId} are searched and the results are merged using a {@link MergingIterator}.
 		COMMENT @return A list of all matching items in this search tree.
 		ARG 1 searchText
+	CLASS MergingUniqueIterator
+		METHOD <init> (Ljava/util/Iterator;Ljava/util/Iterator;Ljava/util/Comparator;)V
+			ARG 3 order
```
Comment (sciwhiz12) [inline data/net/minecraft/client/searchtree/ReloadableSearchTree.mapping]: Parameter names should follow the name of the field they're assigned to.
```suggestion
			ARG 3 orderT
```

File: data/net/minecraft/client/server/IntegratedServer.mapping [hunk 21]
```diff
@@ -1,4 +1,16 @@
...
+		ARG 2 mc
+		ARG 3 registryHolder
+		ARG 4 levelStorageAccess
+		ARG 5 packRepository
+		ARG 6 serverResources
+		ARG 7 worldData
+		ARG 8 session
+		ARG 9 gameProfileRepository
+		ARG 10 gameProfileCache
+		ARG 11 chunkProgressionListenerfactory
```
Comment (sciwhiz12) [inline data/net/minecraft/client/server/IntegratedServer.mapping]: Parameter names should follow the name of the field they're assigned to. (These are from `MinecraftServer.<init>`.)
```suggestion
		ARG 1 serverThread
		ARG 2 minecraft
		ARG 3 registryHolder
		ARG 4 storageSource
		ARG 5 packRepository
		ARG 6 resources
		ARG 7 worldData
		ARG 8 sessionService
		ARG 9 profileRepository
		ARG 10 profileCache
		ARG 11 progressListenerfactory
```

File: data/net/minecraft/client/sounds/MusicManager.mapping [hunk 22]
```diff
@@ -1,4 +1,6 @@
 CLASS net/minecraft/client/sounds/MusicManager
+	METHOD <init> (Lnet/minecraft/client/Minecraft;)V
+		ARG 1 mc
```
Comment (sciwhiz12) [inline data/net/minecraft/client/sounds/MusicManager.mapping]: Parameter names should follow the name of the field they're assigned to.
```suggestion
		ARG 1 minecraft
```

File: data/net/minecraft/client/sounds/SoundBufferLibrary.mapping [hunk 23]
```diff
@@ -1,8 +1,12 @@
 CLASS net/minecraft/client/sounds/SoundBufferLibrary
+	METHOD <init> (Lnet/minecraft/server/packs/resources/ResourceManager;)V
+		ARG 1 manager
```
Comment (sciwhiz12) [inline data/net/minecraft/client/sounds/SoundBufferLibrary.mapping]: Parameter names should follow the name of the field they're assigned to.
```suggestion
		ARG 1 resourceManager
```

File: data/net/minecraft/client/tutorial/Tutorial.mapping [hunk 24]
```diff
@@ -1,4 +1,9 @@
 CLASS net/minecraft/client/tutorial/Tutorial
+	METHOD <init> (Lnet/minecraft/client/Minecraft;)V
+		ARG 1 mc
```
Comment (sciwhiz12) [inline data/net/minecraft/client/tutorial/Tutorial.mapping]: Parameter names should follow the name of the field they're assigned to.
```suggestion
		ARG 1 minecraft
```

File: data/net/minecraft/client/tutorial/Tutorial.mapping [hunk 25]
```diff
@@ -1,4 +1,9 @@
...
+	METHOD addTimedToast (Lnet/minecraft/client/gui/components/toasts/TutorialToast;I)V
+		ARG 1 toast
+		ARG 2 duration
```
Comment (sciwhiz12) [inline data/net/minecraft/client/tutorial/Tutorial.mapping]: Parameter names should follow the name of the field they're (eventually) assigned to.
```suggestion
		ARG 2 durationTicks
```

File: data/net/minecraft/client/tutorial/Tutorial.mapping [hunk 26]
```diff
@@ -17,8 +24,14 @@ CLASS net/minecraft/client/tutorial/Tutorial
...
+		ARG 1 toast
 	METHOD setStep (Lnet/minecraft/client/tutorial/TutorialSteps;)V
 		COMMENT Sets a new step to the tutorial
 		ARG 1 step
 	METHOD start ()V
 		COMMENT Reloads the tutorial step from the game settings
+	CLASS TimedToast
+		METHOD <init> (Lnet/minecraft/client/gui/components/toasts/TutorialToast;I)V
+			ARG 1 toast
+			ARG 2 duration
```
Comment (sciwhiz12) [inline data/net/minecraft/client/tutorial/Tutorial.mapping]: Parameter names should follow the name of the field they're assigned to.
```suggestion
			ARG 2 durationTicks
```

File: data/net/minecraft/commands/arguments/AngleArgument.mapping [hunk 27]
```diff
@@ -2,6 +2,7 @@ CLASS net/minecraft/commands/arguments/AngleArgument
 	METHOD getAngle (Lcom/mojang/brigadier/context/CommandContext;Ljava/lang/String;)F
 		ARG 0 context
 		ARG 1 name
+	METHOD parse (Lcom/mojang/brigadier/StringReader;)Ljava/lang/Object;
```
Comment (sciwhiz12) [inline data/net/minecraft/commands/arguments/AngleArgument.mapping]: Please revert this change and the others similar to it in the same package, as this adds nothing. (This will likely be removed by a `sanitizeData` pass.)

File: data/net/minecraft/commands/arguments/blocks/BlockInput.mapping [hunk 28]
```diff
@@ -7,5 +7,6 @@ CLASS net/minecraft/commands/arguments/blocks/BlockInput
 		ARG 1 level
 		ARG 2 pos
 		ARG 3 flags
+	METHOD test (Ljava/lang/Object;)Z
```
Comment (sciwhiz12) [inline data/net/minecraft/commands/arguments/blocks/BlockInput.mapping]: Please revert this change and the others similar to it in the same package, as this adds nothing. (This will likely be removed by a `sanitizeData` pass.)

File: data/net/minecraft/commands/arguments/coordinates/BlockPosArgument.mapping [hunk 29]
```diff
@@ -2,8 +2,12 @@ CLASS net/minecraft/commands/arguments/coordinates/BlockPosArgument
 	METHOD getLoadedBlockPos (Lcom/mojang/brigadier/context/CommandContext;Ljava/lang/String;)Lnet/minecraft/core/BlockPos;
 		ARG 0 context
 		ARG 1 name
+	METHOD getOrLoadBlockPos (Lcom/mojang/brigadier/context/CommandContext;Ljava/lang/String;)Lnet/minecraft/core/BlockPos;
+		ARG 0 context
+		ARG 1 argumentName
 	METHOD listSuggestions (Lcom/mojang/brigadier/context/CommandContext;Lcom/mojang/brigadier/suggestion/SuggestionsBuilder;)Ljava/util/concurrent/CompletableFuture;
 		ARG 1 context
 		ARG 2 builder
+	METHOD parse (Lcom/mojang/brigadier/StringReader;)Ljava/lang/Object;
```
Comment (sciwhiz12) [inline data/net/minecraft/commands/arguments/coordinates/BlockPosArgument.mapping]: Please revert this change and the others similar to it in the same package, as this adds nothing. (This will likely be removed by a `sanitizeData` pass.)

File: data/net/minecraft/commands/arguments/coordinates/BlockPosArgument.mapping [hunk 30]
```diff
@@ -2,8 +2,12 @@ CLASS net/minecraft/commands/arguments/coordinates/BlockPosArgument
 	METHOD getLoadedBlockPos (Lcom/mojang/brigadier/context/CommandContext;Ljava/lang/String;)Lnet/minecraft/core/BlockPos;
 		ARG 0 context
 		ARG 1 name
+	METHOD getOrLoadBlockPos (Lcom/mojang/brigadier/context/CommandContext;Ljava/lang/String;)Lnet/minecraft/core/BlockPos;
+		ARG 0 context
+		ARG 1 argumentName
```
Comment (sciwhiz12) [inline data/net/minecraft/commands/arguments/coordinates/BlockPosArgument.mapping]: For consistency, this parameter should be named `name`. (It is understood that the `static` methods in argument classes retrieve arguments from the given command context.)
```suggestion
		ARG 1 name
```

File: data/net/minecraft/commands/arguments/item/FunctionArgument.mapping [hunk 31]
```diff
@@ -15,6 +15,7 @@ CLASS net/minecraft/commands/arguments/item/FunctionArgument
 		ARG 0 tag
 	METHOD lambda$static$1 (Ljava/lang/Object;)Lcom/mojang/brigadier/Message;
 		ARG 0 function
+	METHOD parse (Lcom/mojang/brigadier/StringReader;)Ljava/lang/Object;
```
Comment (sciwhiz12) [inline data/net/minecraft/commands/arguments/item/FunctionArgument.mapping]: Please revert this change and the others similar to it in the same package, as this adds nothing. (This will likely be removed by a `sanitizeData` pass.)

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]

**Review by sciwhiz12 — APPROVED** [review]

**Review by Lanse505 — APPROVED** [review]
